internal/libs/autofile/autofile.go

```diff
@@ -170,6 +191,8 @@ func (af *AutoFile) openFile() error {
 	// 	return errors.NewErrPermissionsChanged(file.Name(), fileInfo.Mode(), autoFilePerms)
 	// }
 	af.file = file
+	// Reset .nWrites as this is a fresh file.
+	atomic.StoreUint64(&af.nWrites, 0)
```

Comment:
why are we using atomic writes to write to a value that's always protected by a mutex.

Reply:
This is the fast path, for when it isn't protected by a mutex. I mean I can make it, if y'all prefer that but I've profiled lots of the code in internal/* and mutex locking consumes tons of CPU

Reply:
let's not use an an atomic here given that it's always protected by the mutex.

Reply:
Updated, please take a look and thank you @tychoish!